packages/backend/src/nest/connections-manager/connections-manager.service.ts

```diff
@@ -147,26 +148,74 @@ export class ConnectionsManagerService extends EventEmitter implements OnModuleI
     }
 
     if (this.configOptions.torControlPort) {
-      console.log('launch 1')
+      await this.migrateLevelDb()
       await this.launchCommunityFromStorage()
     }
   }
 
+  /**
+   * Migrate LevelDB when upgrading Quiet for existing communities
+   *
+   * Move data from Redux in the frontend to LevelDB in the backend for existing
+   * communities when upgrading. Hopefully this will make features easier to
+   * test and develop. In order to do this, we need the data to be accessible on
+   * the backend before it's first used. Since the backend starts up
+   * asynchronously, independent of the frontend, we wait for the frontend to
+   * load migration data before launching the community.
+   */
+  public async migrateLevelDb(): Promise<void> {
+    // Empty promise used to wait on a callback below
+    let onDataReceived: () => void
+    const dataReceivedPromise = new Promise<void>((resolve: () => void) => {
+      onDataReceived = resolve
+    })
+    // This is related to a specific migration, perhaps there is a way to
+    // encapsulate this in LocalDbService.
+    const keys = [LocalDBKeys.CURRENT_COMMUNITY_ID, LocalDBKeys.COMMUNITIES]
+    const keysRequired: string[] = []
+
+    for (const key of keys) {
+      if (!(await this.localDbService.exists(key))) {
+        keysRequired.push(key)
+      }
+    }
+
+    this.socketService.on(SocketActionTypes.LOAD_MIGRATION_DATA, async (data: Record<string, any>) => {
+      await this.localDbService.migrate(data)
+      onDataReceived()
+    })
+
+    // Only require migration data for existing communities. We can tell because
+    // they are using the deprecated COMMUNITY key in LevelDB. This is related
+    // to a specific migration. Perhaps we want a more general purpose migration
+    // mechanism, like a table to hold migrations that have already been
+    // applied.
+    if ((await this.localDbService.exists(LocalDBKeys.COMMUNITY)) && keysRequired.length > 0) {
+      this.logger('Migration data required:', keysRequired)
+      this.serverIoProvider.io.emit(SocketActionTypes.MIGRATION_DATA_REQUIRED, keysRequired)
+      await dataReceivedPromise
+    } else {
+      this.logger('Nothing to migrate')
+    }
+  }
```

Comment:
Since we are moving the Community model to the backend, new communities created after this code is released will have the Community model in the backend, however existing communities that upgrade Quiet need to have their community data migrated - that's what this function is for.